[6.0][region-isolation] Move logging in front of NDEBUG to make it easier to triage with no asserts compilers

gottesmm opened the pull request:
Explanation: [region-isolation] Move logging in front of NDEBUG to make it easier to triage with no asserts compilers

This is just an attempt to make it easier to triage reported issues from no-asserts compilers. Today we compile those out, so I need to always compile an asserts compiler to do a basic triage. By moving this in front of NDEBUG, we ensure that we can enable this logging in no-asserts compilers.

Radars:

- rdar://133701799

Original PRs:

- https://github.com/swiftlang/swift/pull/75760
- https://github.com/swiftlang/swift/pull/75810

Risk: Low. This just adds a new macro instead of LLVM_DEBUG that enables/disables the debugging error msgs in the same manner except controlled by a different command line option that is available when asserts are disabled.

gottesmm committed:
[region-isolation] Make logging and debug tooling appear in non-asserts builds.

This will just help me to more quickly triage without needing to compile an
asserts compiler.

(cherry picked from commit 1fbc930cdd604e3c50107b28fa6ba7fe247c211d)

gottesmm committed:
Fix SILBasicBlock::{dump,print}ID so that they are available in no-asserts but just print out NOASSERTS instead of a debug id.

(cherry picked from commit 2e08d6110d386c7b785061ce9a7c0a082b39b8b0)

gottesmm committed:
[region-isolation] Clean up some code now that SILBasicBlock::{dump,print}ID are in front of NDEBUG.

(cherry picked from commit 226e97a5a67ddbf21b1573de95144cd90bdad7e6)

gottesmm commented:
@swift-ci test

hborla commented:
This change is not targeting 6.0. I've converted this to a draft until we have a `release/6.0.0` branch, freeing up `release/6.0` for other potential point release changes.

gottesmm committed:
[region-isolation] Move the region isolation logging decls out of PartitionUtils.h -> RegionIsolation.h

These do not specifically have to do with PartitionUtils... they are really
logging options for the whole infrastructure, so it makes sense to have them in
the a different file.

(cherry picked from commit b477433a20db497bd737f4eecf6497ca39c8c836)

gottesmm committed:
[region-isolation] Tweak the logging to make it easier to quickly find the emitted error when triaging code.

Just trying to improve my triaging speed by making it easier to triage.

(cherry picked from commit a36b3749dc939d47318cf75819d2f5f6678e7c5e)

gottesmm commented:
@swift-ci test

gottesmm commented:
Added a little more logging commits from main that make it quicker to triage.

tbkka reviewed (commented):
Nice!

gottesmm commented:
Didn't need this in the end. Closing.
